WordPress/gutenberg
# Create comments title with simple styling

## What?

Closes #40264

Adds a block to include "X replies to Post" title. 

![Screenshot 2022-04-18 at 15 25 54](https://user-images.githubusercontent.com/37012961/163814768-95886392-3044-4f6a-b53b-4c049c61f8c9.png)


## Why?

This way we keep consistency with the previous Comments list of previous templates. [Mentioned in WP Tavern](https://wptavern.com/gutenberg-13-0-adds-text-selection-across-multiple-blocks-new-stack-group-variation-and-featured-cover-blocks#:~:text=However%2C%20there%20is%20not%20yet%20a%20solution%20for%20the%20%E2%80%9CX%20replies%20to%20Post%E2%80%9D%20title).

## How?

Adding a new block. I have to include:
- [x] Custom wording for `One Response to` and  `X responses to`
- [x] Display with the same styling both in editor and Frontend.
- [x] Test with no comments.
- [x] Add it by default to the Comments Query Loop template.
- [x] Tag selector
- [x] Display everything on Frontend.
- [x] New Block Icon like Archive Title or Post Title. cc @jameskoster 

## Testing Instructions




1.) Add a Comments Query Loop.
2.) Add `Comments Title`
3.) Edit styling.
4.) Check that the frontend displays the styles chosen in the editor.

## Screenshots or screencast
